### PR TITLE
android plugin integration

### PR DESCRIPTION
**Pull request by alfred316**
# Description

Integrate the Android java plugin for extended SDK functionality, like logging app crashes and reporting that to the backend. Including other android lifecycle functionality.

Height Task ID(s) (If applicable): T-9192

## Type of change

> Note: delete the lines that are not applicable and check the boxes (add a capital "X" in the square brackets) for the type of change.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned this PR to myself in GitHub
- [x] I have assigned and notified Reviewers for this PR

### CODE REVIEW

File: Cognitive3DTest/Plugins/Cognitive3D/Source/Cognitive3D/Private/Cognitive3D.cpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ void FAnalyticsCognitive3D::StartupModule()
 	Cognitive3DProvider.Pin()->sensors = new FSensors();
 	Cognitive3DProvider.Pin()->fixationDataRecorder = new FFixationDataRecorder();
 	Cognitive3DProvider.Pin()->gazeDataRecorder = new FGazeDataRecorder();
-	Cognitive3DProvider.Pin()->localCache = MakeShareable(new FLocalCache(FPaths::GeneratedConfigDir()));
+	Cognitive3DProvider.Pin()->localCache = MakeShareable(new FLocalCache(FPaths::Combine(FPaths::ProjectConfigDir(), TEXT("c3dlocal/"))));
```
Comment (matt-manuel) [inline Cognitive3DTest/Plugins/Cognitive3D/Source/Cognitive3D/Private/Cognitive3D.cpp:44]: What's this change for?

Comment (parisacognitive3d) [reply Cognitive3DTest/Plugins/Cognitive3D/Source/Cognitive3D/Private/Cognitive3D.cpp:44]: Previously, in Unreal, we were storing local cache files in the app's data folder. But in Unity, we use a dedicated folder to keep our files separate from the rest of the app's files. So, we applied the same approach here by creating our own folder within the app's data directory.

**Comment by parisacognitive3d** [comment]
> Oh I think I see it better now. The .cpp and .h files are wrappers for the plugin to work with Unreal?

Exactly! These two are essentially used to communicate with the plugin. In Unity, we use `AndroidPlugin.cs` to handle communication between Unity and the Android plugin.

**Review by parisacognitive3d — APPROVED** [review]
LGTM

**Comment by alfred316** [comment]
yes the .cpp and .h AndroidPlugin files contain the AndroidPlugin component class which is added to our c3d bp. This allows our sdk to find the android java plugin class and call functions from it to initialize the plugin, get properties like version number, set up paths for logs in the unreal application, and manage logs when its time to send data to the dashboard.

**Pull request merged**